### DISCUSSION TITLE
Building wheel file using poetry with local wheel file.

### DISCUSSION THREAD

**Discussion by hawksung1** [python-poetry/poetry#9479] [Q&A] [answered]
I working on python 3.10.14 project using poetry.

The project has the following poetry dependency:

```toml
[tool.poetry.dependency]
groundingdino = {file="../external_library/groundingdino-0.1.0-cp310-cp310-linux_x86_64.whl"}
```

During project development, 'poetry install' works fine.

However, after creating a whl file using poetry build, (to deploy this project)
I get an error stating that the whl file is not found in the /tmp path when I try to install the whl file using pip install.
```
No such file or directory: '/home/user/****/groundingdino-0.1.0-cp310-cp310-linux_x86_64.whl'
```

Is there a way to configure the whl file created by poetry build to look for files in the 
'specific relative path' or find whl file in the same path ??

for example.
let project whl name is 'project.whl'

If exist dir like,
/path/project.whl
/path/groundingdino-0.1.0-cp310-cp310-linux_x86_64.whl
or
/path/project.whl
/external_library/groundingdino-0.1.0-cp310-cp310-linux_x86_64.whl

then, following command works. (automatically find groundingdino whl file)
```
pip install project.whl
```

what I want to do is
when I install 'project.whl', set dir to find other whl file. but It finds always in abs path.

**Comment by radoering** [comment, marked as answer]
Afaik, the [standard](https://peps.python.org/pep-0440/#file-urls) (and therefore also Poetry) does not support wheels with relative path dependencies. Since Poetry aims to create valid wheel files, which may be consumed by other tools than Poetry, we cannot deviate from the standard in this case.

In case, you just want to use [pip --find-links](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-f), you have to create a wheel with a name-value-dependency instead of a path dependency, i.e. change your pyproject.toml as follows:

```
[tool.poetry.dependencies]
groundingdino = ">=0.1.0"

[tool.poetry.group.dev.dependencies]
groundingdino = {file="../external_library/groundingdino-0.1.0-cp310-cp310-linux_x86_64.whl"}
```
